### PR TITLE
Ruby done

### PR DESCRIPTION
**Pull request by zeus11011**

### CODE REVIEW

File: ruby/exer1/exer.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    def initialize(inp)
+        @str = inp
+    end
+
+    def parse()
+        @str = @str.split("-")
+        @str = @str.map{|ele| ele.to_f}
+        @str[0] = @str[0].to_i
+    end
+    def getStr
```
Comment (joyrel-vaz) [inline ruby/exer1/exer.rb]: Ruby uses `snake_case` naming convention. Make changes accordingly

Comment (joyrel-vaz) [inline ruby/exer1/exer.rb]: Can use attribute accessor instead of making a method

File: ruby/exer1/Test.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'test/unit'
+require_relative 'exer'
+
+class TestAvg < Test::Unit::TestCase
+    def test_create_average
```
Comment (joyrel-vaz) [inline ruby/exer1/Test.rb]: Write a few more test cases that test for negative cases

File: ruby/exer2/exerc.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+class Arrayfunctions
+    attr_reader :arr
+    def initialize(array)
+        @arr = array
+    end
+
+    def element_at(index)
+        if @arr.length() > index
+            return @arr[index]
```
Comment (joyrel-vaz) [inline ruby/exer2/exerc.rb]: You should use better error handling to guard against out of bound access

File: ruby/exer3/exerc.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    def initialize
+        @hash = Hash.new
+    end 
+
+    def add(key,value)
+        @hash[key] = value
+    end
+
+    def delete_keys_non_int
+        @hash.delete_if {  |key,value|  (key.is_a? Integer) ? false : true }
```
Comment (joyrel-vaz) [inline ruby/exer3/exerc.rb]: Can be refactored

File: ruby/exer3/exerc.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    end
+
+    def delete_keys_odd_int
+        @hash.delete_if {  |key,value|  (key % 2 == 0 ) ? false : true }
```
Comment (joyrel-vaz) [inline ruby/exer3/exerc.rb]: same applies here

File: ruby/exer4/exer.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+class Skip
+  def initialize(arr)
+    @arr = arr
+  end
+
+  def skip_sports(number)
+    if (number > 0 and number < @arr.length)
+      temp = @arr[number..]
```
Comment (joyrel-vaz) [inline ruby/exer4/exer.rb]: This can be simplified, also check what is being returned by the method

File: ruby/exer5/exerc.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+  def transform_file
+    file = File.open("./data.json")
+    @json = JSON.parse(file.read)
+    file.close
+    @json["first name"], @json["last name"] = @json["name"].split(" ")
+    @json.delete("name")
+  end
+
+  def dumpFile
```
Comment (joyrel-vaz) [inline ruby/exer5/exerc.rb]: Class and method names can be renamed for better readability

File: ruby/exer5/exerc.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    begin
+      dump = JSON.dump(@json)
+      @time = Time.new.utc.to_i
+      f = File.new(("./ANISH_%s.json" % @time), "w+")
```
Comment (joyrel-vaz) [inline ruby/exer5/exerc.rb:21]: UTC format does not look like the one in the output file created, check the correct format

File: ruby/exer6/exer.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
+require "base64"
+
+class Encode
+  attr_reader :txt
+
+  def fileread(path)
+    begin
+      file = File.open(path)
+      if file != nil
+        @txt = file.read
```
Comment (joyrel-vaz) [inline ruby/exer6/exer.rb:10]: You don't need to return a value for every method

File: ruby/exer1/exer.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        @str
+    end
+    def ouput()
+        parse()
+        outputStr = @str[0].to_s.rjust(3, "0")+ "-#{"%.2f" % ((@str[1] + @str[2])/ 2)}"
```
Comment (joyrel-vaz) [inline ruby/exer1/exer.rb]: Can be simplified

File: ruby/exer1/exer.rb [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+class Average
+    
+    def initialize(inp)
+        @str = inp
+    end
+
+    def parse()
+        @str = @str.split("-")
+        @str = @str.map{|ele| ele.to_f}
```
Comment (joyrel-vaz) [inline ruby/exer1/exer.rb]: Do not need to parse every array element.

**Review by joyrel-vaz — CHANGES_REQUESTED** [review]

File: ruby/exer6/exer.rb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+      return -1
+    end
+  end
+
+  def encode(name)
+    begin
+      encoded = Base64.encode64("the string that will be base encoded.")
+
+      file = File.open(name + ".txt", "w+")
+      file.write(@txt + " " + encoded)
```
Comment (minaxi98) [inline ruby/exer6/exer.rb]: the given string should be replaced with encoded and not at end of file

**Review by minaxi98 — CHANGES_REQUESTED** [review]
